.github/workflows/ci.yaml

```diff
@@ -27,38 +27,23 @@ jobs:
           attic login chutney https://cache.nixos.asia ${{ secrets.ATTIC_LOGIN_TOKEN }}
           attic use chutney:oss
 
-      - name: Find base commit for incremental build
-        id: find_intermediates_base
+      - name: Find Build Intermediates Base
+        uses: ./.github/actions/find-intermediates-base
         if: github.event_name == 'pull_request'
-        run: |
-          BASE_COMMIT=$(sh .github/workflows/find_intermediates_base.sh)
-          if [ -n "$BASE_COMMIT" ]; then
-            echo "base_commit=$BASE_COMMIT" >> $GITHUB_OUTPUT
-          else
-            echo "No base commit found for incremental build"
-          fi
-
-      - name: Determine build type
-        id: build_type
-        run: |
-          if [ "${{ github.event_name }}" = "pull_request" ] && [ -n "${{ steps.find_intermediates_base.outputs.base_commit }}" ]; then
-            echo "Build type: incremental"
-            echo "build_type=incremental" >> $GITHUB_OUTPUT
-          else
-            echo "Build type: full"
-            echo "build_type=full" >> $GITHUB_OUTPUT
-          fi
+        id: find_build_intermediates
+        with:
+          cache_url: "https://cache.nixos.asia/oss"
 
-      - name: Build
+      - name: Om CI
+        id: om_ci
+        shell: bash
         run: |
-          if [ "${{ steps.build_type.outputs.build_type }}" = "incremental" ]; then
-            WORKTREE_DIR="temp-worktree-${{ steps.find_intermediates_base.outputs.base_commit }}"
-            git worktree add $WORKTREE_DIR ${{ steps.find_intermediates_base.outputs.base_commit }}
-            om ci run -- --override-input "flake/incremental" "path:$WORKTREE_DIR"
-            git worktree remove $WORKTREE_DIR
+          if [ -n "${{ steps.find_build_intermediates.outputs.commit_hash }}" ]; then
+            om ci run -- --override-input "flake/prev" "github:${{ github.repository }}/${{ steps.find_build_intermediates.outputs.commit_hash }}"
```

Comment:
Looks good to me. However, since this action is going to be run on Github anyway, you could output a 2nd var: `flake_ref` or something, so it can be simply used as:

```
om ci run -- --override-input "flake/prev" "${{ steps.find_build_intermediates.outputs.flake_ref }}"
```

Reply:
Also, separate to this PR, we need to fix the issue with having to specifying `flake/`.

Reply:
https://github.com/juspay/haskell-incremental-build-template/pull/5/commits/5b0207ea715a004f392c427423531861d7ccee88

Reply:
> Also, separate to this PR, we need to fix the issue with having to specifying flake/.

Created https://github.com/juspay/omnix/issues/452